scripts/irods/test/test_ichmod.py

```diff
@@ -28,6 +29,183 @@ def test_ichmod_r_with_garbage_path(self):
         self.user.assert_icommand('iget ' + self.admin.session_collection + '/sub_dir1\\\\%/subdir2/file ' + os.path.join(self.user.local_session_dir, ''))
 
+    def test_ichmod_access_levels(self):
+        filename1 = 'test_ichmod_access_levels'
+        filename2 = 'test_access_levels_put_file'
+        local_file_path = os.path.join(self.admin.local_session_dir, filename1)
+        public_collection = os.path.join('/' + self.admin.zone_name, 'home', 'public')
+
+        test_object = os.path.join(public_collection, filename1)
+        test_collection_for_put = os.path.join(public_collection, 'test_ichmod_collection_put')
+        test_collection_for_access = os.path.join(public_collection, 'test_ichmod_collection_access')
+        test_collection_for_create = os.path.join(test_collection_for_put, 'new_collection')
+
+        user_file_path = os.path.join(self.user.local_session_dir, filename2)
+        user_iget_file = os.path.join(self.user.local_session_dir, filename1)
+
+        coll_access_path = os.path.join(test_collection_for_put, filename1)
+
+        class Access(Enum):
+            NULL            = 1
+            READ_METADATA   = 2
+            READ_OBJECT     = 3
+            CREATE_METADATA = 4
+            MODIFY_METADATA = 5
+            DELETE_METADATA = 6
+            CREATE_OBJECT   = 7
+            MODIFY_OBJECT   = 8
+            DELETE_OBJECT   = 9
+            OWN             = 10
+
+        # The access_specifier can be a string, or a list of strings
+        # if aliases exist. All aliases will be tested
+        access_specifiers = {
+            Access.NULL            : "null",
+            Access.READ_METADATA   : "read_metadata",
+            Access.READ_OBJECT     : ["read_object", "read"],
+            Access.CREATE_METADATA : "create_metadata",
+            Access.MODIFY_METADATA : "modify_metadata",
+            Access.DELETE_METADATA : "delete_metadata",
+            Access.CREATE_OBJECT   : "create_object",
+            Access.MODIFY_OBJECT   : ["modify_object", "write"],
+            Access.DELETE_OBJECT   : "delete_object",
+            Access.OWN             : "own"
+        }
+
+        # Here the test commands we test with every access level are defined:
+        # 'cmd' is the command to run with assert_icommand
+        # the 'checks' list contains an access level range and the output of the command to match
+        # in the format (<min_level>, <max_level>, <arguments>)
+        # *arguments will be passed to the assert_icommands function
+        #
+        #
+        test_commands = {
+            # Test that iget needs READ OBJECT
+            'iget': {
+                'cmd': ['iget' , '-f', test_object, user_iget_file],
+                'checks': [
+                    (Access.NULL, Access.READ_METADATA,  ['STDERR', 'USER_INPUT_PATH_ERR']),
+                    (Access.READ_METADATA, Access.OWN,   ['STDERR', ''])
```

Comment:
should this be READ_OBJECT?

Reply:
I think you're correct. And `STDERR` should be removed because `ichmod` doesn't produce output on success.

Reply:
well, this is `iget`...

Reply:
Whoops. You're right. Ignore me lol

Reply:
I think this should be `Access.READ_OBJECT`, yes. This test currently passes - I *think* - because it checks for anything or nothing being on stderr, which will always be the case. So, the suggested edit by @korydraughn should be the right thing to do regardless.

Reply:
done - now incorporated.